### ISSUE TITLE
[feature request] make it possible to get supported diagram types via REST

### ISSUE THREAD

**Issue by FibreFoX** [yuzutech/kroki#746]
**Description**

Right now when visiting a self-hosted instance of kroki, it is not possible to get the supported diagram types via REST as some kind of JSON-response. All you can get is a HTML-page that lists the types (via `HelloHandler`). This makes it hard to auto-detect which diagram type is supported.

**Proposal**

In order to check which diagram types are supported, there needs to be some REST endpoint to get this information.
Inside the server you already have some kind of registry: https://github.com/yuzutech/kroki/blob/d4294d10e70afd25db646466b759eb173cff58a4/server/src/main/java/io/kroki/server/Server.java#L100

**Outcome**

Not only would it enable the CLI to have some kind of diagram type discovery (which is already reported/requested at https://github.com/yuzutech/kroki-cli/issues/15), but it also would support the integration of other tools, which need to check if the diagram-type could be generated by kroki.

**Comment by Mogztter** [comment]
I guess you could use https://kroki.io/health API?

Please keep in mind that this API (currently) returns all supported diagram types. It won't actually check if a diagram type is actually available. For instance, if you don't start the Mermaid companion container, the `/health` API will still return `mermaid`.

If we want to return the actual list then we will need to setup a mechanism to check at runtime which diagram types are actually available. 
It might also be useful to get which output formats are supported per diagram type.

**Comment by FibreFoX** [comment]
Yes, I could use https://kroki.io/health ... although from the name it does suggest a different use-case.

Regarding the validity of that list, thats another topic. To solve that, my first idea was to have some kind of "bootstrap process" on first server-boot, like calling each "companion container" once with a small example, but it is more a discovery but a "health"-check. What do you think @Mogztter , should this be another issue to be created?
